src/extensions/warnings.ts

```diff
@@ -0,0 +1,53 @@
+import * as marked from "marked";
+import * as clc from "cli-color";
+
+import { ExtensionVersion, RegistryLaunchStage } from "./extensionsApi";
+import { printSourceDownloadLink } from "./displayExtensionInfo";
+import { logPrefix } from "./extensionsHelper";
+import { getTrustedPublishers } from "./resolveSource";
+import { promptOnce } from "../prompt";
+import * as utils from "../utils";
+
+function displayEAPWarning(publisherId: string, sourceDownloadUri: string, githubLink?: string) {
+  const publisherNameLink = githubLink ? `[${publisherId}](${githubLink})` : publisherId;
+  const warningMsg = `This extension is in preview and is built by a developer in the [Extensions Publisher Early Access Program](http://bit.ly/firex-provider). Its functionality might change in backward-incompatible ways. Since this extension isn't built by Firebase, reach out to ${publisherNameLink} with questions about this extension.`;
```

Comment:
nit: try to stick to 80 chars/line

Reply:
Usually I agree - however for lines that just declare template literals/strings, I find that this is counterproductive + leads to harder to read code with unnecessary string concatenation.

Reply:
Yeah the ts support for multiline string is not great... but still I find that to be better than having to looking for a horizontal scroll bar and dragging it...
don't feel too strongly about it so I'll leave it up to you